ENIGMAsystem/SHELL/Universal_System/Resources/AssetArray.h

```diff
@@ -121,8 +184,14 @@ class AssetArray {
   size_t size() const { return assets_.size(); }
   bool exists(int id) const { return (id >= 0 && size_t(id) < size() && !assets_[id].isDestroyed()); }
 
+  T* data() { return assets_.data(); }

```

Comment:
This is dangerous. Only implement this if we *really* need to.

Reply:
need it for [] unless you want me copy pasting -left logic into 12 functions

Reply:
Where? Why aren't they just using `operator[]`? We can't do bounds checking (even in debug mode) when we return a raw pointer that may or may not be inside our buffer. The way we use it now is unlikely to really throw people, though, I suppose.

Reply:
I implement operator [] and it calls data()[index]...

Reply:
You implemented that within the OffsetVector itself. This is exposing OffsetVector::data() to clients of AssetArray, which might reasonably assume that the allocated block contains all elements between min_id and max_id. Which is true when LEFT ≤ 0, which is all we're using it for right now, so... fine? But this is a leaky abstraction.